dashboard/app/controllers/lti/v1/account_linking_controller.rb

```diff
@@ -23,7 +23,10 @@ def finish_link
 
       # POST /lti/v1/account_linking/link_email
       def link_email
-        head :bad_request unless PartialRegistration.in_progress?(session)
+        unless PartialRegistration.in_progress?(session)
+          head :bad_request and return
```

Comment:
I think `head` might be safe from this issue, at least according to [the docs](https://api.rubyonrails.org/v7.1.3.4/classes/ActionController/Head.html#method-i-head) that say it "**Returns** a response that has no content (merely headers)"

Reply:
Ah good call on reading the docs. ChatGPT told me I should add an explicit return here, but I think you're right. Also, this isn't the line of code causing the error, so I'll put this back as it was and only add the `and return` to the redirect on line 50.